### PR TITLE
PTX-18757 Upgrade PX and Operator in air-gapped env

### PR DESCRIPTION
**Pull request by nikolaypopov**
Upgrade PX and Operator in air-gapped env

### CODE REVIEW

File: drivers/volume/portworx/portworx.go [hunk 1]
```diff
@@ -2996,6 +3026,138 @@ func configurePxReleaseManifestEnvVars(origEnvVarList []corev1.EnvVar, specGenUR
...
+	}
+
+	// Download PX versions URL content
+	pxVersionsFile := path.Join("/versions")
+	cmd := fmt.Sprintf("curl -o %s %s", pxVersionsFile, pxVersionURL)
+	strOut, strErr, err := osutils.ExecShell(cmd)
+	if err != nil {
+		return fmt.Errorf("Failed to download versions file from URL [%s], Err: %v %v", pxVersionURL, strErr, err)
+	}
+	log.Debug(strOut)
```
Comment (mborodin-px) [inline drivers/volume/portworx/portworx.go:3069]: remove debug line

File: tests/basic/upgrade_test.go [hunk 2]
```diff
@@ -2,11 +2,12 @@ package tests
...
 import (
 	"fmt"
-	"github.com/portworx/sched-ops/k8s/core"
-	"github.com/portworx/torpedo/pkg/osutils"
 	"strings"
 	"time"
 
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/portworx/torpedo/pkg/osutils"
+
```
Comment (mborodin-px) [inline tests/basic/upgrade_test.go:10]: remove empty line

**Review by mborodin-px — APPROVED** [review]
just minor comments

File: drivers/volume/portworx/portworx.go (hunk 1 above)
Comment (nikolaypopov) [reply drivers/volume/portworx/portworx.go]: Done

File: tests/basic/upgrade_test.go (hunk 2 above)
Comment (nikolaypopov) [reply tests/basic/upgrade_test.go:10]: Done

**Review by CharudathGopal — APPROVED** [review]

File: drivers/volume/portworx/portworx.go [hunk 3]
```diff
@@ -2996,6 +3023,139 @@ func configurePxReleaseManifestEnvVars(origEnvVarList []corev1.EnvVar, specGenUR
...
+	// Get custom registry configmap configmap
+	_, err := core.Instance().GetConfigMap(pxCustomRegistryConfigmapName, pxNamespace)
+	if k8serrors.IsNotFound(err) {
+		return false
+	}
+	return true
+}
+
+// isPxVersionsConfigMapExist( return true or false based on wether px-versions cm exists or not
+func isPxVersionsConfigMapExist(pxNamespace string) bool {
```
Comment (jainsmit) [inline drivers/volume/portworx/portworx.go:3037]: why two methods to check two different configMap ? we can have single method and passing the configMap name as parameter ..

**Review by jainsmit — APPROVED** [review]
minor comment !!

File: drivers/volume/portworx/portworx.go (hunk 3 above)
Comment (nikolaypopov) [reply drivers/volume/portworx/portworx.go:3037]: Good catch! Added the second one last minute when I found a bug and didn't realize I am looking for same thing in both functions!

**Pull request merged**